packages/rocketchat-lib/server/methods/leaveRoom.js

```diff
@@ -24,7 +24,13 @@ Meteor.methods({
 		if (RocketChat.authz.hasRole(user._id, 'owner', room._id)) {
 			const numOwners = RocketChat.authz.getUsersInRole('owner', room._id).fetch().length;
 			if (numOwners === 1) {
-				throw new Meteor.Error('error-you-are-last-owner', 'You are the last owner. Please set new owner before leaving the room.', { method: 'leaveRoom' });
+				if (room.usernames.length > 1) {
+					throw new Meteor.Error('error-you-are-last-owner', 'You are the last owner. Please set new owner before leaving the room.', { method: 'leaveRoom' });
+				} else {
+					RocketChat.models.Messages.removeByRoomId(rid);
```

Comment:
We should warn the user that leaving an empty room will destroy the room, together with its history. Leaving a room usually doesn't imply "delete its history", so this change in action could lead to confusion and data loss.